code/.devcontainer/Dockerfile

```diff
@@ -1,12 +1,25 @@
+# Copyright (c) 2021-2023 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
 FROM mcr.microsoft.com/devcontainers/typescript-node:18-bookworm
 
 ADD install-vscode.sh /root/
 RUN /root/install-vscode.sh
 
 RUN git config --system codespaces-theme.hide-status 1
 
+USER 0
+
+# install yarn and node-gyp
+RUN npm install --global yarn@1 node-gyp@9
```

Comment:
could you clarify why we need these changes for the upstream dockerfile?

Reply:
so that node-gyp is installed and we're consistent across the dockerfile?